ocf_data_sampler/load/satellite.py

```diff
@@ -1,59 +1,230 @@
 """Satellite loader."""
-import numpy as np
+
+import logging
+import os
+import re
+from typing import List, Optional
+
+import dask
+import icechunk
 import xarray as xr
-from xarray_tensorstore import open_zarr
 
 from ocf_data_sampler.load.utils import (
     check_time_unique_increasing,
     get_xr_data_array_from_xr_dataset,
     make_spatial_coords_increasing,
 )
 
-from .open_tensorstore_zarrs import open_zarrs
+logger = logging.getLogger(__name__)
 
+# Optimal values from research, now hardcoded as per Sol's feedback.
+OPTIMAL_BLOCK_SIZE_MB = 64
+OPTIMAL_THREADS = 2
 
-def open_sat_data(zarr_path: str | list[str]) -> xr.DataArray:
-    """Lazily opens the zarr store and validates data types.
-
-    Args:
-      zarr_path: Cloud URL or local path pattern, or list of these. If GCS URL,
-                 it must start with 'gs://'
-    """
-    # Open the data
-    if isinstance(zarr_path, list | tuple):
-        ds = open_zarrs(zarr_path, concat_dim="time")
-    else:
-        ds = open_zarr(zarr_path)
+def _setup_optimal_environment():
+    """Apply optimization settings for cloud data streaming."""
+    logger.debug("Applying optimization settings for cloud streaming...")
 
-    check_time_unique_increasing(ds.time)
+    os.environ["GCSFS_CACHE_TIMEOUT"] = "3600"
+    os.environ["GCSFS_BLOCK_SIZE"] = str(OPTIMAL_BLOCK_SIZE_MB * 1024 * 1024)
+    os.environ["GCSFS_DEFAULT_CACHE_TYPE"] = "readahead"
+    os.environ["GOOGLE_CLOUD_DISABLE_GRPC"] = "true"
 
-    ds = ds.rename(
+    dask.config.set(
         {
-            "variable": "channel",
-            "time": "time_utc",
-        },
+            "distributed.worker.memory.target": 0.7,
+            "array.chunk-size": "512MB",
+            "distributed.comm.compression": None,
+            "distributed.worker.threads": OPTIMAL_THREADS,
+        }
     )
+    logger.debug("Optimization environment configured successfully")
+
+def _parse_icechunk_path(path: str) -> dict:
```

Comment:
Since this is just parsing the zarr path now, the function name should change to reflect it isn't icechunk-specific.